packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,232 @@
+
+  log(`Extracted ${docRefs.length} attachments`);
+  const docRefBundleEntries = docRefs.map(dr => ({ resource: dr }));
+  const collectionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "collection",
+    entry: docRefBundleEntries,
+  };
+
+  const transactionBundle = convertCollectionBundleToTransactionBundle({
+    fhirBundle: collectionBundle,
+  });
+
+  if (transactionBundle.entry?.length && fhirUrl && s3BucketName.length) {
+    log(`Starting uploads. ${JSON.stringify(transactionBundle)}`);
+    const fhirApi = makeFhirApi(cxId, fhirUrl);
+    await executeWithNetworkRetries(async () => await fhirApi.executeBatch(transactionBundle), {
+      log,
+    });
+
+    executeAsynchronously(uploadDetails, async uploadParams => {
+      await s3Utils.uploadFile(uploadParams);
+    });
+  }
+  log(`Done...`);
+}
+
+function buildDocumentReferenceDraft(
+  patientId: string,
+  extensions: Extension[]
+): DocumentReference {
+  return {
+    resourceType: "DocumentReference",
+    status: "current",
+    docStatus: "final",
+    subject: {
+      reference: `Patient/${patientId}`,
+      type: "Patient",
+    },
+    extension: extensions,
+  };
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getDetails(document: CdaOriginalText | undefined): {
+  fileB64Contents: string | undefined;
+  mimeType: string | undefined;
+} {
+  const fileB64Contents = document?.["#text"];
+  const mimeType = document?._mediaType;
+
+  return {
+    fileB64Contents,
+    mimeType,
+  };
+}
+
+function buildDocumentReference(
+  patientId: string,
+  extensions: Extension[],
+  act: ConcernActEntryAct
+) {
+  const docRef = buildDocumentReferenceDraft(patientId, extensions);
+  const identifiers = getIdentifiers(act.id);
+  const date = getDate(act.effectiveTime);
+  const type = getType(act.code);
+
+  docRef.id = uuidv4();
+
+  if (identifiers.length) docRef.identifier = identifiers;
+  if (type) {
+    docRef.type = type;
+    if (type.text) docRef.description = type.text;
+  }
+  if (date) docRef.date = date;
+
+  return docRef;
+}
+
+function getIdentifiers(id: CdaInstanceIdentifier | undefined): Identifier[] {
+  return [
+    {
+      ...(id?._root && { system: id._root }),
+      ...(id?._extension && { value: id._extension }),
+    },
+  ];
+}
+
+function getSourceExtension(medicalSource: string | undefined): Extension | undefined {
+  if (isMedicalDataSource(medicalSource)) {
+    if (medicalSource === MedicalDataSource.CAREQUALITY) return cqExtension;
+    if (medicalSource === MedicalDataSource.COMMONWELL) return cwExtension;
+  }
+  return undefined;
+}
+
+function getDate(time: EffectiveTimeLowHigh | undefined): string | undefined {
+  if (time?.low?._value) return buildDayjs(normalizeDateFromFhir(time.low._value)).toISOString();
+  if (time?.high?._value) return buildDayjs(normalizeDateFromFhir(time.high._value)).toISOString();
+  return undefined;
+}
+
+function normalizeDateFromFhir(dateString: string) {
```

Comment:
should we turn the input string into dayJs and then just format it? Would be cleaner and more stable I think.

Reply:
maybe dayJs can't handle timezones?

Reply:
i tried giving dayjs the string that XML has it in, i.e. "20230913230000+0000", and it's invalid. I tried using `.format` as well, and no luck 😒

Reply:
word